File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofModels/SportLeague.swift
import Foundation

public enum SportLeague: String, Codable, CaseIterable, Sendable, Hashable {
    case nfl
    case cfb
    case nba
    case ncaab
    case mlb

    public var displayName: String {
        switch self {
        case .nfl: "NFL"
        case .cfb: "College Football"
        case .nba: "NBA"
        case .ncaab: "College Basketball"
        case .mlb: "MLB"
        }
    }

    public var sfSymbol: String {
        switch self {
        case .nfl, .cfb: "football.fill"
        case .nba, .ncaab: "basketball.fill"
        case .mlb: "baseball.fill"
        }
    }
}

File: wagerproof-android-native/core/models/src/main/kotlin/com/wagerproof/core/models/SportLeague.kt
package com.wagerproof.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Global sport enum (iOS `SportLeague`). Wire values are the lowercase raws. */
@Serializable
enum class SportLeague(val raw: String) {
    @SerialName("nfl") NFL("nfl"),
    @SerialName("cfb") CFB("cfb"),
    @SerialName("nba") NBA("nba"),
    @SerialName("ncaab") NCAAB("ncaab"),
    @SerialName("mlb") MLB("mlb");

    val displayName: String
        get() = when (this) {
            NFL -> "NFL"
            CFB -> "College Football"
            NBA -> "NBA"
            NCAAB -> "College Basketball"
            MLB -> "MLB"
        }

    // iOS SF Symbol name, kept for cross-platform parity; Android maps it to its own icon resource.
    val sfSymbol: String
        get() = when (this) {
            NFL, CFB -> "football.fill"
            NBA, NCAAB -> "basketball.fill"
            MLB -> "baseball.fill"
        }
}
